packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -92,13 +92,15 @@ export class Dropdown extends BaseComponent<IDropdownProps, IDropdownState> {
           onClick={ this._onDropdownClick }
           aria-expanded={ isOpen ? 'true' : 'false' }
           role='combobox'
+          aria-live={ disabled || isOpen ? 'off' : 'assertive' }
```

Comment:
I agree with you that other component may interrupt,  but I think the option text should be read out immediately once selected option is changed rather than waiting until it's idle.